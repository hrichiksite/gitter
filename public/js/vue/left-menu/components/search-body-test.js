const mount = require('../../__test__/vuex-mount');
const { default: SearchBody } = require('./search-body.vue');

const {
  createSerializedRoomFixture,
  createSerializedMessageSearchResultFixture
} = require('../../__test__/fixture-helpers');

const ROOM_SEARCH_RESULT_ROOM1 = createSerializedRoomFixture('community/room1');
const ROOM_SEARCH_RESULT_ROOM2 = createSerializedRoomFixture('community/room2');
const ROOM_SEARCH_ROOM_MAP_FIXTURE = {
  [ROOM_SEARCH_RESULT_ROOM1.id]: ROOM_SEARCH_RESULT_ROOM1,
  [ROOM_SEARCH_RESULT_ROOM2.id]: ROOM_SEARCH_RESULT_ROOM2
};
const ROOM_SEARCH_RESULT_FIXTURE = [ROOM_SEARCH_RESULT_ROOM1.id, ROOM_SEARCH_RESULT_ROOM2.id];

const MESSAGE_SEARCH_RESULT_FIXTURE = [
  createSerializedMessageSearchResultFixture(),
  createSerializedMessageSearchResultFixture()
];

describe('search-body', () => {
  it('matches snapshot', () => {
    const { wrapper } = mount(SearchBody);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('room search repo loading matches snapshot', () => {
    const { wrapper } = mount(SearchBody, {}, store => {
      store.state.search.repo.loading = true;
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('room search repo error matches snapshot', () => {
    const { wrapper } = mount(SearchBody, {}, store => {
      store.state.search.repo.error = true;
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('room search repo results matches snapshot', () => {
    const { wrapper } = mount(SearchBody, {}, store => {
      store.state.roomMap = ROOM_SEARCH_ROOM_MAP_FIXTURE;
      store.state.search.repo.results = ROOM_SEARCH_RESULT_FIXTURE;
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('room search room loading matches snapshot', () => {
    const { wrapper } = mount(SearchBody, {}, store => {
      store.state.search.room.loading = true;
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('room search room error matches snapshot', () => {
    const { wrapper } = mount(SearchBody, {}, store => {
      store.state.search.room.error = true;
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('room search room results matches snapshot', () => {
    const { wrapper } = mount(SearchBody, {}, store => {
      store.state.roomMap = ROOM_SEARCH_ROOM_MAP_FIXTURE;

      store.state.search.room.results = ROOM_SEARCH_RESULT_FIXTURE;
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('room search people loading matches snapshot', () => {
    const { wrapper } = mount(SearchBody, {}, store => {
      store.state.search.people.loading = true;
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('room search people error matches snapshot', () => {
    const { wrapper } = mount(SearchBody, {}, store => {
      store.state.search.people.error = true;
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('room search people results matches snapshot', () => {
    const { wrapper } = mount(SearchBody, {}, store => {
      store.state.roomMap = ROOM_SEARCH_ROOM_MAP_FIXTURE;

      store.state.search.people.results = ROOM_SEARCH_RESULT_FIXTURE;
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('room search combined results matches snapshot', () => {
    const { wrapper } = mount(SearchBody, {}, store => {
      const repo1 = createSerializedRoomFixture('community/repo1');
      const room1 = createSerializedRoomFixture('community/room1');
      const person1 = createSerializedRoomFixture('person1');

      store.state.roomMap = {
        [repo1.id]: repo1,
        [room1.id]: room1,
        [person1.id]: person1
      };

      store.state.search.repo.results = [repo1.id];
      store.state.search.room.results = [room1.id];
      store.state.search.people.results = [person1.id];
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('deduplicates room search combined results matches snapshot', () => {
    const { wrapper } = mount(SearchBody, {}, store => {
      const room1 = createSerializedRoomFixture('community/repo1');

      store.state.roomMap = {
        [room1.id]: room1
      };

      store.state.search.current.results = [room1.id];
      store.state.search.repo.results = [room1.id];
      store.state.search.room.results = [room1.id];
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('message search loading matches snapshot', () => {
    const { wrapper } = mount(SearchBody, {}, store => {
      store.state.search.message.loading = true;
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('message search error matches snapshot', () => {
    const { wrapper } = mount(SearchBody, {}, store => {
      store.state.search.message.error = true;
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('message search results matches snapshot', () => {
    const { wrapper } = mount(SearchBody, {}, store => {
      store.state.search.message.results = MESSAGE_SEARCH_RESULT_FIXTURE;
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('calls store action "fetchRoomSearchResults" and "fetchMessageSearchResults" after inputting', () => {
    const searchValue = 'needtofindthisthing';
    const { wrapper, stubbedActions } = mount(SearchBody, { searchImmediately: true }, store => {
      store.state.search.searchInputValue = searchValue;
    });

    wrapper.find({ ref: 'search-input' }).trigger('input');

    expect(stubbedActions.fetchRoomSearchResults).toHaveBeenCalled();
    expect(stubbedActions.fetchMessageSearchResults).toHaveBeenCalled();
  });
});
