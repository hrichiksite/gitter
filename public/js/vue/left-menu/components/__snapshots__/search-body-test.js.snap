// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search-body deduplicates room search combined results matches snapshot 1`] = `
<div
  class="search-body-root"
>
  <h2
    class="search-body-title"
  >
    Search
  </h2>
   
  <div
    class="search-input-wrapper"
  >
    <input
      class="search-input"
      placeholder="What are you looking for?"
    />
  </div>
   
  <ul
    class="syntax-list"
  >
    <li
      class="syntax-list-item"
    >
      <code>
        from:username
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        text:@username
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        sent:[2020-06-08 TO 2020-06-09]
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        sent:[2020-06-09T16:17-03:00 TO 2020-06-09T16:21-03:00]
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        text:@(username1 OR username2)
      </code>
       
      <br />
       
      <code>
        text:@(username1 AND username2)
      </code>
    </li>
  </ul>
   
  <h2
    class="search-body-title"
  >
    
    Rooms & People
    
    <!---->
  </h2>
   
  <ul
    class="room-search-list"
  >
    <list-item-stub
      item="[object Object]"
      type="all"
    />
  </ul>
   
  <!---->
   
  <!---->
   
  <!---->
   
  <h2
    class="search-body-title"
  >
    
    Chat messages
    
    <!---->
  </h2>
   
  <div
    class="search-result-empty-message"
  >
    
    No message search results...
  
  </div>
</div>
`;

exports[`search-body matches snapshot 1`] = `
<div
  class="search-body-root"
>
  <h2
    class="search-body-title"
  >
    Search
  </h2>
   
  <div
    class="search-input-wrapper"
  >
    <input
      class="search-input"
      placeholder="What are you looking for?"
    />
  </div>
   
  <ul
    class="syntax-list"
  >
    <li
      class="syntax-list-item"
    >
      <code>
        from:username
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        text:@username
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        sent:[2020-06-08 TO 2020-06-09]
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        sent:[2020-06-09T16:17-03:00 TO 2020-06-09T16:21-03:00]
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        text:@(username1 OR username2)
      </code>
       
      <br />
       
      <code>
        text:@(username1 AND username2)
      </code>
    </li>
  </ul>
   
  <h2
    class="search-body-title"
  >
    
    Rooms & People
    
    <!---->
  </h2>
   
  <div
    class="search-result-empty-message"
  >
    
    No room search results...
  
  </div>
   
  <!---->
   
  <!---->
   
  <!---->
   
  <h2
    class="search-body-title"
  >
    
    Chat messages
    
    <!---->
  </h2>
   
  <div
    class="search-result-empty-message"
  >
    
    No message search results...
  
  </div>
</div>
`;

exports[`search-body message search error matches snapshot 1`] = `
<div
  class="search-body-root"
>
  <h2
    class="search-body-title"
  >
    Search
  </h2>
   
  <div
    class="search-input-wrapper"
  >
    <input
      class="search-input"
      placeholder="What are you looking for?"
    />
  </div>
   
  <ul
    class="syntax-list"
  >
    <li
      class="syntax-list-item"
    >
      <code>
        from:username
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        text:@username
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        sent:[2020-06-08 TO 2020-06-09]
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        sent:[2020-06-09T16:17-03:00 TO 2020-06-09T16:21-03:00]
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        text:@(username1 OR username2)
      </code>
       
      <br />
       
      <code>
        text:@(username1 AND username2)
      </code>
    </li>
  </ul>
   
  <h2
    class="search-body-title"
  >
    
    Rooms & People
    
    <!---->
  </h2>
   
  <div
    class="search-result-empty-message"
  >
    
    No room search results...
  
  </div>
   
  <!---->
   
  <!---->
   
  <!---->
   
  <h2
    class="search-body-title"
  >
    
    Chat messages
    
    <!---->
  </h2>
   
  <div
    class="search-result-error-message"
  >
    
    Error fetching messages
  
  </div>
</div>
`;

exports[`search-body message search loading matches snapshot 1`] = `
<div
  class="search-body-root"
>
  <h2
    class="search-body-title"
  >
    Search
  </h2>
   
  <div
    class="search-input-wrapper"
  >
    <input
      class="search-input"
      placeholder="What are you looking for?"
    />
  </div>
   
  <ul
    class="syntax-list"
  >
    <li
      class="syntax-list-item"
    >
      <code>
        from:username
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        text:@username
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        sent:[2020-06-08 TO 2020-06-09]
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        sent:[2020-06-09T16:17-03:00 TO 2020-06-09T16:21-03:00]
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        text:@(username1 OR username2)
      </code>
       
      <br />
       
      <code>
        text:@(username1 AND username2)
      </code>
    </li>
  </ul>
   
  <h2
    class="search-body-title"
  >
    
    Rooms & People
    
    <!---->
  </h2>
   
  <div
    class="search-result-empty-message"
  >
    
    No room search results...
  
  </div>
   
  <!---->
   
  <!---->
   
  <!---->
   
  <h2
    class="search-body-title"
  >
    
    Chat messages
    
    <loading-spinner-stub />
  </h2>
   
  <div
    class="search-result-empty-message"
  >
    
    No message search results...
  
  </div>
</div>
`;

exports[`search-body message search results matches snapshot 1`] = `
<div
  class="search-body-root"
>
  <h2
    class="search-body-title"
  >
    Search
  </h2>
   
  <div
    class="search-input-wrapper"
  >
    <input
      class="search-input"
      placeholder="What are you looking for?"
    />
  </div>
   
  <ul
    class="syntax-list"
  >
    <li
      class="syntax-list-item"
    >
      <code>
        from:username
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        text:@username
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        sent:[2020-06-08 TO 2020-06-09]
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        sent:[2020-06-09T16:17-03:00 TO 2020-06-09T16:21-03:00]
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        text:@(username1 OR username2)
      </code>
       
      <br />
       
      <code>
        text:@(username1 AND username2)
      </code>
    </li>
  </ul>
   
  <h2
    class="search-body-title"
  >
    
    Rooms & People
    
    <!---->
  </h2>
   
  <div
    class="search-result-empty-message"
  >
    
    No room search results...
  
  </div>
   
  <!---->
   
  <!---->
   
  <!---->
   
  <h2
    class="search-body-title"
  >
    
    Chat messages
    
    <!---->
  </h2>
   
  <ul
    class="message-search-list"
  >
    <search-body-message-result-item-stub
      messagesearchresult="[object Object]"
    />
    <search-body-message-result-item-stub
      messagesearchresult="[object Object]"
    />
  </ul>
</div>
`;

exports[`search-body room search combined results matches snapshot 1`] = `
<div
  class="search-body-root"
>
  <h2
    class="search-body-title"
  >
    Search
  </h2>
   
  <div
    class="search-input-wrapper"
  >
    <input
      class="search-input"
      placeholder="What are you looking for?"
    />
  </div>
   
  <ul
    class="syntax-list"
  >
    <li
      class="syntax-list-item"
    >
      <code>
        from:username
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        text:@username
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        sent:[2020-06-08 TO 2020-06-09]
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        sent:[2020-06-09T16:17-03:00 TO 2020-06-09T16:21-03:00]
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        text:@(username1 OR username2)
      </code>
       
      <br />
       
      <code>
        text:@(username1 AND username2)
      </code>
    </li>
  </ul>
   
  <h2
    class="search-body-title"
  >
    
    Rooms & People
    
    <!---->
  </h2>
   
  <ul
    class="room-search-list"
  >
    <list-item-stub
      item="[object Object]"
      type="all"
    />
    <list-item-stub
      item="[object Object]"
      type="all"
    />
    <list-item-stub
      item="[object Object]"
      type="all"
    />
  </ul>
   
  <!---->
   
  <!---->
   
  <!---->
   
  <h2
    class="search-body-title"
  >
    
    Chat messages
    
    <!---->
  </h2>
   
  <div
    class="search-result-empty-message"
  >
    
    No message search results...
  
  </div>
</div>
`;

exports[`search-body room search people error matches snapshot 1`] = `
<div
  class="search-body-root"
>
  <h2
    class="search-body-title"
  >
    Search
  </h2>
   
  <div
    class="search-input-wrapper"
  >
    <input
      class="search-input"
      placeholder="What are you looking for?"
    />
  </div>
   
  <ul
    class="syntax-list"
  >
    <li
      class="syntax-list-item"
    >
      <code>
        from:username
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        text:@username
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        sent:[2020-06-08 TO 2020-06-09]
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        sent:[2020-06-09T16:17-03:00 TO 2020-06-09T16:21-03:00]
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        text:@(username1 OR username2)
      </code>
       
      <br />
       
      <code>
        text:@(username1 AND username2)
      </code>
    </li>
  </ul>
   
  <h2
    class="search-body-title"
  >
    
    Rooms & People
    
    <!---->
  </h2>
   
  <div
    class="search-result-empty-message"
  >
    
    No room search results...
  
  </div>
   
  <!---->
   
  <!---->
   
  <div
    class="search-result-error-message"
  >
    
    Error fetching people
  
  </div>
   
  <h2
    class="search-body-title"
  >
    
    Chat messages
    
    <!---->
  </h2>
   
  <div
    class="search-result-empty-message"
  >
    
    No message search results...
  
  </div>
</div>
`;

exports[`search-body room search people loading matches snapshot 1`] = `
<div
  class="search-body-root"
>
  <h2
    class="search-body-title"
  >
    Search
  </h2>
   
  <div
    class="search-input-wrapper"
  >
    <input
      class="search-input"
      placeholder="What are you looking for?"
    />
  </div>
   
  <ul
    class="syntax-list"
  >
    <li
      class="syntax-list-item"
    >
      <code>
        from:username
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        text:@username
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        sent:[2020-06-08 TO 2020-06-09]
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        sent:[2020-06-09T16:17-03:00 TO 2020-06-09T16:21-03:00]
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        text:@(username1 OR username2)
      </code>
       
      <br />
       
      <code>
        text:@(username1 AND username2)
      </code>
    </li>
  </ul>
   
  <h2
    class="search-body-title"
  >
    
    Rooms & People
    
    <loading-spinner-stub />
  </h2>
   
  <div
    class="search-result-empty-message"
  >
    
    No room search results...
  
  </div>
   
  <!---->
   
  <!---->
   
  <!---->
   
  <h2
    class="search-body-title"
  >
    
    Chat messages
    
    <!---->
  </h2>
   
  <div
    class="search-result-empty-message"
  >
    
    No message search results...
  
  </div>
</div>
`;

exports[`search-body room search people results matches snapshot 1`] = `
<div
  class="search-body-root"
>
  <h2
    class="search-body-title"
  >
    Search
  </h2>
   
  <div
    class="search-input-wrapper"
  >
    <input
      class="search-input"
      placeholder="What are you looking for?"
    />
  </div>
   
  <ul
    class="syntax-list"
  >
    <li
      class="syntax-list-item"
    >
      <code>
        from:username
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        text:@username
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        sent:[2020-06-08 TO 2020-06-09]
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        sent:[2020-06-09T16:17-03:00 TO 2020-06-09T16:21-03:00]
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        text:@(username1 OR username2)
      </code>
       
      <br />
       
      <code>
        text:@(username1 AND username2)
      </code>
    </li>
  </ul>
   
  <h2
    class="search-body-title"
  >
    
    Rooms & People
    
    <!---->
  </h2>
   
  <ul
    class="room-search-list"
  >
    <list-item-stub
      item="[object Object]"
      type="all"
    />
    <list-item-stub
      item="[object Object]"
      type="all"
    />
  </ul>
   
  <!---->
   
  <!---->
   
  <!---->
   
  <h2
    class="search-body-title"
  >
    
    Chat messages
    
    <!---->
  </h2>
   
  <div
    class="search-result-empty-message"
  >
    
    No message search results...
  
  </div>
</div>
`;

exports[`search-body room search repo error matches snapshot 1`] = `
<div
  class="search-body-root"
>
  <h2
    class="search-body-title"
  >
    Search
  </h2>
   
  <div
    class="search-input-wrapper"
  >
    <input
      class="search-input"
      placeholder="What are you looking for?"
    />
  </div>
   
  <ul
    class="syntax-list"
  >
    <li
      class="syntax-list-item"
    >
      <code>
        from:username
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        text:@username
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        sent:[2020-06-08 TO 2020-06-09]
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        sent:[2020-06-09T16:17-03:00 TO 2020-06-09T16:21-03:00]
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        text:@(username1 OR username2)
      </code>
       
      <br />
       
      <code>
        text:@(username1 AND username2)
      </code>
    </li>
  </ul>
   
  <h2
    class="search-body-title"
  >
    
    Rooms & People
    
    <!---->
  </h2>
   
  <div
    class="search-result-empty-message"
  >
    
    No room search results...
  
  </div>
   
  <div
    class="search-result-error-message"
  >
    
    Error fetching repos
  
  </div>
   
  <!---->
   
  <!---->
   
  <h2
    class="search-body-title"
  >
    
    Chat messages
    
    <!---->
  </h2>
   
  <div
    class="search-result-empty-message"
  >
    
    No message search results...
  
  </div>
</div>
`;

exports[`search-body room search repo loading matches snapshot 1`] = `
<div
  class="search-body-root"
>
  <h2
    class="search-body-title"
  >
    Search
  </h2>
   
  <div
    class="search-input-wrapper"
  >
    <input
      class="search-input"
      placeholder="What are you looking for?"
    />
  </div>
   
  <ul
    class="syntax-list"
  >
    <li
      class="syntax-list-item"
    >
      <code>
        from:username
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        text:@username
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        sent:[2020-06-08 TO 2020-06-09]
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        sent:[2020-06-09T16:17-03:00 TO 2020-06-09T16:21-03:00]
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        text:@(username1 OR username2)
      </code>
       
      <br />
       
      <code>
        text:@(username1 AND username2)
      </code>
    </li>
  </ul>
   
  <h2
    class="search-body-title"
  >
    
    Rooms & People
    
    <loading-spinner-stub />
  </h2>
   
  <div
    class="search-result-empty-message"
  >
    
    No room search results...
  
  </div>
   
  <!---->
   
  <!---->
   
  <!---->
   
  <h2
    class="search-body-title"
  >
    
    Chat messages
    
    <!---->
  </h2>
   
  <div
    class="search-result-empty-message"
  >
    
    No message search results...
  
  </div>
</div>
`;

exports[`search-body room search repo results matches snapshot 1`] = `
<div
  class="search-body-root"
>
  <h2
    class="search-body-title"
  >
    Search
  </h2>
   
  <div
    class="search-input-wrapper"
  >
    <input
      class="search-input"
      placeholder="What are you looking for?"
    />
  </div>
   
  <ul
    class="syntax-list"
  >
    <li
      class="syntax-list-item"
    >
      <code>
        from:username
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        text:@username
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        sent:[2020-06-08 TO 2020-06-09]
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        sent:[2020-06-09T16:17-03:00 TO 2020-06-09T16:21-03:00]
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        text:@(username1 OR username2)
      </code>
       
      <br />
       
      <code>
        text:@(username1 AND username2)
      </code>
    </li>
  </ul>
   
  <h2
    class="search-body-title"
  >
    
    Rooms & People
    
    <!---->
  </h2>
   
  <ul
    class="room-search-list"
  >
    <list-item-stub
      item="[object Object]"
      type="all"
    />
    <list-item-stub
      item="[object Object]"
      type="all"
    />
  </ul>
   
  <!---->
   
  <!---->
   
  <!---->
   
  <h2
    class="search-body-title"
  >
    
    Chat messages
    
    <!---->
  </h2>
   
  <div
    class="search-result-empty-message"
  >
    
    No message search results...
  
  </div>
</div>
`;

exports[`search-body room search room error matches snapshot 1`] = `
<div
  class="search-body-root"
>
  <h2
    class="search-body-title"
  >
    Search
  </h2>
   
  <div
    class="search-input-wrapper"
  >
    <input
      class="search-input"
      placeholder="What are you looking for?"
    />
  </div>
   
  <ul
    class="syntax-list"
  >
    <li
      class="syntax-list-item"
    >
      <code>
        from:username
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        text:@username
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        sent:[2020-06-08 TO 2020-06-09]
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        sent:[2020-06-09T16:17-03:00 TO 2020-06-09T16:21-03:00]
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        text:@(username1 OR username2)
      </code>
       
      <br />
       
      <code>
        text:@(username1 AND username2)
      </code>
    </li>
  </ul>
   
  <h2
    class="search-body-title"
  >
    
    Rooms & People
    
    <!---->
  </h2>
   
  <div
    class="search-result-empty-message"
  >
    
    No room search results...
  
  </div>
   
  <!---->
   
  <div
    class="search-result-error-message"
  >
    
    Error fetching rooms
  
  </div>
   
  <!---->
   
  <h2
    class="search-body-title"
  >
    
    Chat messages
    
    <!---->
  </h2>
   
  <div
    class="search-result-empty-message"
  >
    
    No message search results...
  
  </div>
</div>
`;

exports[`search-body room search room loading matches snapshot 1`] = `
<div
  class="search-body-root"
>
  <h2
    class="search-body-title"
  >
    Search
  </h2>
   
  <div
    class="search-input-wrapper"
  >
    <input
      class="search-input"
      placeholder="What are you looking for?"
    />
  </div>
   
  <ul
    class="syntax-list"
  >
    <li
      class="syntax-list-item"
    >
      <code>
        from:username
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        text:@username
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        sent:[2020-06-08 TO 2020-06-09]
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        sent:[2020-06-09T16:17-03:00 TO 2020-06-09T16:21-03:00]
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        text:@(username1 OR username2)
      </code>
       
      <br />
       
      <code>
        text:@(username1 AND username2)
      </code>
    </li>
  </ul>
   
  <h2
    class="search-body-title"
  >
    
    Rooms & People
    
    <loading-spinner-stub />
  </h2>
   
  <div
    class="search-result-empty-message"
  >
    
    No room search results...
  
  </div>
   
  <!---->
   
  <!---->
   
  <!---->
   
  <h2
    class="search-body-title"
  >
    
    Chat messages
    
    <!---->
  </h2>
   
  <div
    class="search-result-empty-message"
  >
    
    No message search results...
  
  </div>
</div>
`;

exports[`search-body room search room results matches snapshot 1`] = `
<div
  class="search-body-root"
>
  <h2
    class="search-body-title"
  >
    Search
  </h2>
   
  <div
    class="search-input-wrapper"
  >
    <input
      class="search-input"
      placeholder="What are you looking for?"
    />
  </div>
   
  <ul
    class="syntax-list"
  >
    <li
      class="syntax-list-item"
    >
      <code>
        from:username
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        text:@username
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        sent:[2020-06-08 TO 2020-06-09]
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        sent:[2020-06-09T16:17-03:00 TO 2020-06-09T16:21-03:00]
      </code>
    </li>
     
    <li
      class="syntax-list-item"
    >
      <code>
        text:@(username1 OR username2)
      </code>
       
      <br />
       
      <code>
        text:@(username1 AND username2)
      </code>
    </li>
  </ul>
   
  <h2
    class="search-body-title"
  >
    
    Rooms & People
    
    <!---->
  </h2>
   
  <ul
    class="room-search-list"
  >
    <list-item-stub
      item="[object Object]"
      type="all"
    />
    <list-item-stub
      item="[object Object]"
      type="all"
    />
  </ul>
   
  <!---->
   
  <!---->
   
  <!---->
   
  <h2
    class="search-body-title"
  >
    
    Chat messages
    
    <!---->
  </h2>
   
  <div
    class="search-result-empty-message"
  >
    
    No message search results...
  
  </div>
</div>
`;
